tests/integration/dns_test.py

```diff
@@ -573,3 +573,39 @@ def test_invalid_dns_option_with_value(static_dns):
     )
     with pytest.raises(NmstateValueError):
         libnmstate.apply(desired_state)
+
+
+@pytest.fixture
+def static_dns_with_options(static_dns):
+    desired_state = {
+        DNS.KEY: {
+            DNS.CONFIG: {
+                DNS.OPTIONS: ["debug", "rotate"],
+            }
+        },
+    }
+    libnmstate.apply(desired_state)
+
+
+def test_remove_all_dns_options(static_dns_with_options):
+    desired_state = {
+        DNS.KEY: {
+            DNS.CONFIG: {
+                DNS.OPTIONS: [],
+            }
+        },
+    }
+    libnmstate.apply(desired_state)
+
+    current_state = libnmstate.show()
+    assert DNS.OPTIONS not in current_state[DNS.KEY][DNS.CONFIG]
+    assert current_state[DNS.KEY][DNS.CONFIG][DNS.SERVER]
```

Comment:
maybe instead of asserting the existence of DNS server and search, can we assert that they are actually preserved by adding a variable holding the state before applying the DNS option.

Reply:
Test case updated with assert on old server and search not changed.